### DISCUSSION TITLE
Should we use camelCase of snake_case

### DISCUSSION THREAD

**Discussion by bonface221** [prisma/prisma#24926] [Q&A] [answered]
### Question

Should we use camelCase of snake_case in our prisma schema

### Expected behavior (optional)

Should we use camelCase of snake_case

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @bonface221 👋 

Thank you for raising this question.

By default, Prisma uses camelCase for field names in the schema and Pascal Case for Models. This is more natural for JavaScript/TypeScript developers as it aligns with common naming conventions in these languages. If your database schema uses snake_case and you want to maintain this convention in your Prisma schema, you can use the `@map` and `@@map` attributes to map Prisma model and field names to the corresponding database names. This allows you to keep your Prisma Client API in camelCase while the database remains in snake_case. For Example:
```prisma
model MyUser {  
  userId    Int     @id @default(autoincrement()) @map("user_id")  
  firstName String? @map("first_name")  
  lastName  String  @unique @map("last_name")  
  
  @@map("my_user")  
}
```
In conclusion, while Prisma defaults to camelCase, you can customize your schema to use snake_case if that aligns better with your database conventions.

See this [documentation](https://www.prisma.io/docs/orm/reference/prisma-schema-reference#naming-conventions) on naming conventions in Prisma ORM.
